common/notification/notification.ts

```diff
@@ -147,6 +147,27 @@ export async function create(notification: Prisma.notificationCreateInput) {
     return result;
 }
 
+export async function deleteOne(id: NotificationID) {
+    if (!(await getNotification(id, true))) {
+        throw new Error(`Unknown Notification (${id})`);
+    } else if ((await prisma.concrete_notification.count({ where: { notificationID: id } })) > 0) {
+        throw new Error('Cannot delete a notification which has concrete notifications.');
+    }
+
+    // If there is a related message_translation for this notification it has to be deleted first due to foreign key constraint
+    if ((await prisma.message_translation.count({ where: { notificationId: id } })) > 0) {
```

Comment:
check is useless, can just delete(...) (which does nothing anyways if no message translation exists)